libcloudforensics/providers/gcp/internal/gke.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Google Kubernetes Engine functionalities."""
+
+from typing import TYPE_CHECKING, Any, Dict, Optional
+from libcloudforensics.providers.gcp.internal import common
+
+if TYPE_CHECKING:
+  import googleapiclient
+
+class GoogleKubernetesEngine:
+  """Class to call Google Kubernetes Engine (GKE) APIs.
+
+  https://cloud.google.com/kubernetes-engine/docs/reference/rest
+
+  Attributes:
+    gke_api_client: Client to interact with GKE APIs.
+    project_id: Google Cloud project ID.
+  """
+  GKE_API_VERSION = 'v1'
+
+  def __init__(self, project_id: Optional[str] = None) -> None:
+    """Initialize the GoogleCloudStorage object.
+
+    Args:
+      project_id (str): Optional. Google Cloud project ID.
+    """
+
+    self.gke_api_client = None
+    self.project_id = project_id
+
+  def GkeApi(self) -> 'googleapiclient.discovery.Resource':
+    """Gets a Google Container service object.
+
+    https://container.googleapis.com/$discovery/rest?version=v1
+
+    Returns:
+      googleapiclient.discovery.Resource: A Google Container service object.
+    """
+
+    if self.gke_api_client:
+      return self.gke_api_client
+    self.gke_api_client = common.CreateService(
+        'container', self.GKE_API_VERSION)
+    return self.gke_api_client
+
+  def GetCluster(self, name: str) -> Dict[str, Any]:
+    """ Gets the details of a specific cluster.
+
+    Args:
+      name (string): The name (project, location, cluster) of the cluster to retrieve.
```

Comment:
I think in other parts of the codebase we use `str` instead of `string` but this is being really picky, feel free to ignore